--- a/src/plugins/platforms/wayland/qwaylandwindow.cpp
+++ b/src/plugins/platforms/wayland/qwaylandwindow.cpp
@@ -654,9 +654,9 @@
         // make sure isExposed is false during the next event dispatch
         mExposed = false;
         sendExposeEvent(QRect());
-        resetSurfaceRole();
         mSurface->attach(nullptr, 0, 0);
         mSurface->commit();
+        resetSurfaceRole();
     }
 }
 
